### ISSUE TITLE
Android Rich Media Ads not Focusable

### ISSUE THREAD

**Issue by IcarusComplex** [invertase/react-native-google-mobile-ads#230]
Hello!

We run many rich media ads for our partners on our platforms. Since migrating over to this project rich media ads with `<input>` tags in the creatives have not been focusable, i.e. the keyboard never appears and we cannot capture user information. A longpress does show the paste options for the field.

Using version 8.1.0 as reference. This is due to line 165 in `ReactNativeGoogleMobileAdsBannerAdViewManager.java`, `adView.setDescendantFocusability(ViewGroup.FOCUS_BLOCK_DESCENDANTS);`. Any reason why the focusability of descendants is being disabled? 

In the meantime we've created a fork and commented out this line so the `<input>` fields in our creatives are working as intended. Any other way we can achieve the same behaviour?

Thank you!

**Comment by DoctorJohn** [comment]
Hey, just some context:

`adView.setDescendantFocusability(ViewGroup.FOCUS_BLOCK_DESCENDANTS);` was added in 835bdec6aff843e2ef21250137927fb004750aeb to prevent a crash described [here](https://github.com/facebook/react-native/issues/32649#issuecomment-990968256).

**Comment by mikehardy** [comment]
Interesting! Great investigation @DoctorJohn 
I've posted in the forum asking for an update since it appears to be a native SDK issue https://groups.google.com/g/google-admob-ads-sdk/c/EsLEaoRjyNM/m/3mCjddGcAAAJ

It's also possible they have resolved it, who knows?

@IcarusComplex this appears to be a very long-running issue, which leaves me open to a workaround if it's important to you and done cleanly

Specifically if you proposed a PR that allowed for setting a potentially dangerous parameter (that is, named like "allowInputInAdsWithCrashRisk" or something) so you could boolean toggle it on and conditionally not set that block descendants config on the view, it could work. And you/we may discover that it no longer crashes and the bug was inadvertently fixed in some SDK update.

Or if you're fine with patch-package you can just roll with that.

**Comment by IcarusComplex** [comment]
@DoctorJohn Thank you for the context, exactly what I was looking for. 

@mikehardy Thank you. Given how important this is to our commercial teams I'm going to create a fork doing something similar to what you've described so I'll be able to turn this on/off in prod via remote config incase this does introduce a crash.

Whether it's stable or introduces a crash I'll provide some feedback here. If its stable, I'll look into creating a PR that implements a clean workaround.

**Comment by IcarusComplex** [comment]
@mikehardy - just an update, finally had this in a release about a week ago. Since then we're having some native crashes on screens with Ads, not much to go on in the crashlytics logs unfortunately. Will share a more detailed update when I find a gap, for now just an update that I'm still monitoring this.

**Comment by IcarusComplex** [comment]
Finally getting around to writing an update here. Unfortunately, it's not much.

Removing `ReactNativeGoogleMobileAdsBannerAdViewManager.java, adView.setDescendantFocusability(ViewGroup.FOCUS_BLOCK_DESCENDANTS);` 100% still introduces a severe crash on Android. 

This affected a substantial amount of our users, and was immediately resolved upon pushing a remote change to reactivate the above line. I'm including the crashlog as was reported in Firebase Crashlytics, maybe someone else can make any kind of sense of it. 

I'm closing this issue for now. 

```
Crashed: Thread: SIGABRT  0x0000000000000000
#00 pc 0x1a70e libc.so 
#01 pc 0x2be32 libhwui.so 
#02 pc 0x2be32 libhwui.so 
#03 pc 0x2be32 libhwui.so 
#04 pc 0x65a3 liblog.so 
#05 pc 0x96b05 libhwui.so 
#06 pc 0x65be liblog.so 
#07 pc 0x96b707fe  
#08 pc 0x96c1032c  
#09 pc 0x96c46f5a  
#10 pc 0x96b748b1  
#11 pc 0x96c1032c  
#12 pc 0x52457 libc.so 
#13 pc 0x977ad382  
#14 pc 0x977ad382  
#15 pc 0x97722781  
#16 pc 0x96c10f9e

**Comment by mikehardy** [comment]
@IcarusComplex the upstream SDK team appears to be waiting for some sort of share-able reproducible example, otherwise there may be no forward progress: https://groups.google.com/g/google-admob-ads-sdk/c/EsLEaoRjyNM/m/n-MIWADQAAAJ

**Comment by IcarusComplex** [comment]
I've been speaking with google mobile ads sdk directly. I've never been able to reproduce this crash locally in RN, nor has any of our QA or other devs so its a tricky one to pin down for them. Literally all we've got is the crash rate increases dramatically when I make the descendants focusable, and completely recovers when reintroducing the line above.

Will feedback here again if I ever make some progress with the google guys.

**Comment by IcarusComplex** [comment]
@mikehardy Just an update, this issue seems to be resolved now. I've deployed a patch package to production allowing us to toggle this remotely. 

We've been serving focusable Ads for the last few days and crash rates have remained stable. 

Currently using:
react-native-google-mobile-ads 10.3.2
googleMobileAds: 22.0.0

We received an update from the google team that Unity SDK had reported this issue resolved after the release of 22.0.0. Figured we'd give it another go, and so far so good.
